pkg/resource/deploy/source_eval.go

```diff
@@ -993,11 +997,14 @@ func (rm *resmon) RegisterResource(ctx context.Context,
 
 		// Invoke the provider's Construct RPC method.
 		options := plugin.ConstructOptions{
-			Aliases:              aliases,
-			Dependencies:         dependencies,
-			Protect:              protect,
-			PropertyDependencies: propertyDependencies,
-			Providers:            providerRefs,
+			Aliases:      aliases,
+			Dependencies: dependencies,
+			Protect:      protect,
+			Providers:    providerRefs,
+		}
+		if !hasOutputs {
+			// Only include property dependencies when we don't have output values in the properties.
+			options.PropertyDependencies = propertyDependencies
```

Comment:
Could we instead iterate the property dependencies and wrap each corresponding value with an `Output` that carries its deps? Then we could leave the `PropertyDependencies` out altogether from the call to `Construct`, I think.